indico/modules/events/editing/models/revisions.py

```diff
@@ -176,3 +181,15 @@ def is_submitter_revision(self):
     @property
     def is_editor_revision(self):
         return self.type.is_editor_action
+
+
+@listens_for(orm.mapper, 'after_configured', once=True)
+def _mappers_configured():
+    from indico.modules.events.editing.models.comments import EditingRevisionComment
+    query = (select([db.func.greatest(db.func.max(EditingRevisionComment.modified_dt),
+                                      db.func.max(EditingRevisionComment.created_dt),
+                                      EditingRevision.modified_dt, EditingRevision.created_dt)])
+             .where(EditingRevisionComment.revision_id == EditingRevision.id)
+             .correlate_except(EditingRevisionComment)
+             .scalar_subquery())
```

Comment:
one issue with this query is that it doesn't take into account if a revision has been undone, since we don't store the timestamp for that action..

i can only think of two solutions for this
1. replace the is_undone column with a `undone_dt` column and adding that to the `db.func.greatest` of this query
2. adding a marker to the timestamp (such as "U") to indicate that the last revision has been undone. then we can check if the user's timestamp also contains the "U". this is a bit hacky but it avoids the migration

what do you think?

Reply:
will this timestamp ever decrease? because right now you check `moment(lastUpdate).isBefore(editable.lastUpdateDt))`.

if you could replace this with something like `lastUpdate !== editable.lastUpdateDt`, then I'd go for option 2, but call it something else that does not sound like something that's a valid timestamp (and then adding the `U` for undone would be fine)

Reply:
true! it shouldnt ever decrease i dont think.. i'll try that thanks